integration/oplog/basic_oplog_test.go

```diff
@@ -0,0 +1,186 @@
+			_, err := coll.InsertOne(ctx, bson.D{{"_id", int64(1)}, {"foo", "bar"}})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(t, actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			expected, err := types.NewDocument(
+				"op", "i",
+				"ns", ns,
+				"o", must.NotFail(types.NewDocument("_id", int64(1), "foo", "bar")),
+				"ts", must.NotFail(actual.Get("ts")).(types.Timestamp),
+				"v", expectedV,
+			)
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+		})
+
+		t.Run("MultipleDocuments", func(t *testing.T) {
+			_, err := coll.InsertMany(ctx, []any{
+				bson.D{{"_id", int64(2)}, {"foo2", "bar2"}},
+				bson.D{{"_id", int64(3)}, {"foo3", "bar3"}},
+			})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(t, actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			// The last oplog entry should be the last insert.
+			expected, err := types.NewDocument(
+				"op", "i",
+				"ns", ns,
+				"o", must.NotFail(types.NewDocument("_id", int64(3), "foo3", "bar3")),
+				"ts", must.NotFail(actual.Get("ts")).(types.Timestamp),
+				"v", expectedV,
+			)
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+		})
+	})
+
+	t.Run("Delete", func(t *testing.T) {
+		t.Run("SingleDocument", func(t *testing.T) {
+			_, err := coll.DeleteOne(ctx, bson.D{{"_id", int64(1)}})
+			require.NoError(t, err)
+
+			var lastOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&lastOplogEntry)
+			require.NoError(t, err)
+
+			actual := integration.ConvertDocument(t, lastOplogEntry)
+			unsetUnusedOplogFields(t, actual)
+			actualKeys := actual.Keys()
+			assert.ElementsMatch(t, expectedKeys, actualKeys)
+
+			expected, err := types.NewDocument(
+				"op", "d",
+				"ns", ns,
+				"o", must.NotFail(types.NewDocument("_id", int64(1))),
+				"ts", must.NotFail(actual.Get("ts")).(types.Timestamp),
+				"v", expectedV,
+			)
+			require.NoError(t, err)
+			assert.Equal(t, expected, actual)
+
+			// If an entry to delete is not found, expect oplog not to be written.
+			_, err = coll.DeleteOne(ctx, bson.D{{"_id", "non-existent"}})
+			require.NoError(t, err)
+
+			var newOplogEntry bson.D
+			err = local.Collection("oplog.rs").FindOne(ctx, bson.D{{"ns", ns}}, opts).Decode(&newOplogEntry)
+			require.NoError(t, err)
+			assert.Equal(t, lastOplogEntry, newOplogEntry)
+		})
+
+		/*t.Run("MultipleDocuments", func(t *testing.T) {
```

Comment:
This is the case I don't understand. I don't see oplog data for `DeleteMany` while the documents are deleted.

I also tried a few different approaches manually, but it doesn't produce oplog data for `DeleteMany`.

```
db.mycoll.insertMany([{_id: 2, v:2}, {_id: 3, v: 3}]) // oplog entries created as expected
db.mycoll.deleteMany({"_id":{$gte: 2}}) // documents dropped successfully, but no new entries in oplog

db.mycoll.insertMany([{_id: 2, v:2}, {_id: 3, v: 3}]) // oplog entries created
db.mycoll.deleteMany({"_id":2}) // one doc dropped, oplog entry created
db.mycoll.deleteMany({"_id":{$gte: 2}}) // one doc dropped, oplog entry created

db.mycoll.insertMany([{_id: 2, v:2}, {_id: 3, v: 3}]) // oplog entries created
db.mycoll.deleteMany({"_id":{$in: [2, 3]}}) // docs dropped, oplog entries NOT created

db.mycoll.insertMany([{_id: 2, v:2}, {_id: 3, v: 3}]) // oplog entries created
db.mycoll.deleteMany({"_id":{$in: [2]}}) // oplog enries created
```

So, it seems to me that as soon as the result of `deleteMany` has more than one document deleted, oplog entries are not created for these deleted docs. But it's rather a bug than a feature...

Reply:
Is it worth to create issue to investigate it? If so, please do and consider copying this commented code there or adding a skip rule, rather than keeping it commented here.

Reply:
@b1ron PTAL